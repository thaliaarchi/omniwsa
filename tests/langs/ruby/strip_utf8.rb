puts "Ruby #{RUBY_VERSION}"
puts "UTF-8 whitespace characters for String#strip:"
(0..0x10FFFF).each do |c|
  next if c.between?(0xD800, 0xDFFF)
  s = c.chr(Encoding::UTF_8)
  both = "#{s}x#{s}"
  stripped = both.strip
  if stripped != both
    direction = if stripped == "x"
      "on both sides"
    elsif stripped == "x#{s}"
      "only on the left"
    elsif stripped == "#{s}x"
      "only on the right"
    else
      "BUG"
    end
    puts "U+#{c.to_s(16).upcase.rjust(4, '0')} (#{s.inspect}) is stripped #{direction}"
  end
end
