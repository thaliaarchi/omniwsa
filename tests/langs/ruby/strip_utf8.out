Ruby 3.3.6
UTF-8 whitespace characters for String#strip:
U+0000 ("\u0000") is stripped on both sides
U+0009 ("\t") is stripped on both sides
U+000A ("\n") is stripped on both sides
U+000B ("\v") is stripped on both sides
U+000C ("\f") is stripped on both sides
U+000D ("\r") is stripped on both sides
U+0020 (" ") is stripped on both sides
